modules/foreman_host_power.py

```diff
@@ -0,0 +1,167 @@
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from ansible.module_utils.foreman_helper import ForemanEntityAnsibleModule
+ANSIBLE_METADATA = {'metadata_version': '1.0',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: foreman_host_power
+short_description: Manage Foreman hosts power state
+description:
+  - "Manage power state of Foreman host"
+  - "This beta version can start and stop an existing foreman host and question the current power state."
+  - "Uses https://github.com/SatelliteQE/nailgun"
+version_added: "2.7"
+author:
+  - "Bernhard Hopfenmueller (@Fobhep) ATIX AG"
+requirements:
+  - "nailgun >= 0.29.0"
+  - "ansible >= 2.3"
+options:
+  server_url:
+    description:
+      - URL of Foreman server
+    required: true
+  username:
+    description:
+      - Username on Foreman server
+    required: true
+  password:
+    description:
+      - Password for user accessing Foreman server
+    required: true
+  verify_ssl:
+    description:
+      - Verify SSL of the Foreman server
+    required: false
+    default: true
+    type: bool
+  name:
+    description:
+      - name of host
+    required: true
+  domain_name:
+    description:
+      - name of host's domain
+    required: true
+  power_state:
+    description: Desired power state
+    type: list
+    choices:
+      - on
+      - off
+      - state
+'''
+
+EXAMPLES = '''
+- name: "Switch a host on"
+  foreman_host_power:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "test-host"
+    domain_name: "domain.test"
+    state: on
+
+- name: "Switch a host off"
+  foreman_host_power:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "test-host"
+    domain_name: "domain.test"
+    state: off
+
+- name: "Query host power state"
+  foreman_host_power:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    name: "test-host"
+    domain_name: "domain.test"
+    state: state
+    register: result
+- debug:
+    msg: "Host power state is {{ result.power_state }}"
+
+
+'''
+
+RETURN = '''
+power_state:
+    description: current power state of host
+    returned: always
+    type: string
+    sample: "off"
+ '''
+
+try:
+    from nailgun.entities import (
+        Host,
+    )
+
+    from ansible.module_utils.ansible_nailgun_cement import (
+        find_host,
+        naildown_power_state,
+        query_power_state,
+    )
+except ImportError:
+    pass
+
+
+# This is the only true source for names (and conversions thereof)
+name_map = {
+    'name': 'name',
+    'domain_name': 'domain_name',
+}
```

Comment:
Is this actually used anywhere? If so, i can't see it :-)

Reply:
yeah - no that is superfluous :)

Reply:
fixed